### PR TITLE
HWP systematics

### PR DESCRIPTION
**Pull request by nraffuzz**
This PR has been created to address HWP systematics

### CODE REVIEW

**Comment by ziotom78** [comment]
Hi @nraffuzz , something weird happened, as it seems that the whole file `litebird_sim/hwp_sys/hwp_sys.py` was changed. Was there a CR/LF conversion that happened underway?

**Comment by nraffuzz** [comment]
Hi @ziotom78, I saw that, when comparing the old with the new version, two empty lines where missing so that in the end everything was shifted. Probably due to that, everything was counted as something new. 
However, what it has actually changed is the substitution of `sim.parameter_file` with `sim.parameters`, from line 85 to 99 in `hwp_sys.py`, in order to correctly read the dictionary.

**Comment by ziotom78** [comment]
Sorry for the delay in answering you. I found the culprit: you saved the file using DOS line endings instead of Unix (the ones used in the original file). I converted them back, and now the diff now looks reasonable: https://github.com/litebird/litebird_sim/pull/232/files

May you please have a check that the diff is what you expect?

**Comment by nraffuzz** [comment]
Yes, I confirm that the difference is what I expected, thanks for finding the issue!

**Comment by ziotom78** [comment]
Hi, it seems that something weird has happened again… Several files have had their attributes changed: for instance, `bin/git/init-hooks` has its permission flags changed from 100755 to 100644 (and several others). WIth these changes, the PR reports that ~120 files are modified by this commit. Could you please have a check?

**Comment by nraffuzz** [comment]
Hi, I didn't specifically get what went wrong, and of course we didn't want to change permission in those files. How can we solve this?

**Comment by ziotom78** [comment]
I was able to fix permissions, and now the PR lists just 20 files that have been changed. Does this look reasonable?

**Review by ziotom78 — COMMENTED** [review]
Hi @nraffuzz and @sgiardie , very nice work! I have put a couple of tiny comments, have a look and see if you think they are reasonable. No objections to merge on my side.

File: litebird_sim/hwp_sys/hwp_sys.py [hunk 1]
```diff
@@ -38,11 +41,618 @@ def _dBodTth(nu):
...
+                np.dot(
+                    pauli_basis[i],
+                    np.dot(jones, np.dot(pauli_basis[j], np.conjugate(jones).T)),
+                )
+            )
+            # Sanity check to ensure the formula operates correctly and
+            # does not yield any imaginary components.
+            # Mueller-matrix elements should be real.
+            if np.imag(Mij) > 1e-9:
+                print("Discarding the unnecessary imaginary part!")
```
[the remark this answers is not in the document]
Comment (nraffuzz) [reply litebird_sim/hwp_sys/hwp_sys.py]: Done!

File: litebird_sim/bandpasses.py [hunk 2]
```diff
@@ -69,6 +79,8 @@ class BandPassInfo:
 
     bandcenter_ghz: float = 0.0
     bandwidth_ghz: float = 0.0
+    bandlow_ghz: float = 0.0
+    bandhigh_ghz: float = 0.0
```
[the remark this answers is not in the document]
Comment (sgiardie) [reply litebird_sim/bandpasses.py]: Thanks for the suggestion, we set it to None as a default like this:
`bandlow_ghz: Union[float, None] = None`
 `bandhigh_ghz: Union[float, None] = None`

File: litebird_sim/bandpasses.py [hunk 3]
```diff
@@ -90,8 +102,13 @@ def __post_init__(self):
...
         # before  and after the edges
-        bandrange = (self.f0 - self.bandwidth_ghz, self.f1 + self.bandwidth_ghz)
-        self.freqs_ghz = np.linspace(bandrange[0], bandrange[1], self.nsamples_inband)
+        if self.bandlow_ghz == 0.0:
+            self.bandlow_ghz = self.bandcenter_ghz - self.bandwidth_ghz
+        if self.bandhigh_ghz == 0.0:
+            self.bandhigh_ghz = self.bandcenter_ghz + self.bandwidth_ghz
+        self.freqs_ghz = np.linspace(
+            self.bandlow_ghz, self.bandhigh_ghz, self.nsamples_inband
+        )
```
[the remark this answers is not in the document]
Comment (sgiardie) [reply litebird_sim/bandpasses.py]: Thanks again for this suggestion! I think it's very important and has pushed us to add two checks:
- the first one is that the central frequency has to be > 0
- the second one is that the bandpass edges `f0 = self.bandcenter_ghz - self.bandwidth_ghz / 2, f1 =        self.bandcenter_ghz + self.bandwidth_ghz / 2`, fall within the `bandlow_ghz` and `bandhigh_ghz` values. Indeed the `bandlow/high_ghz` parameters represent the limits of the frequency array used for the computation of the bandpass, while the edges encompass the freqs over which the transmission is close to 1

We also decided to add other warnings in `hwp_sys/bandpass_template_module.py` when some parameters are not defined. All of this in general pushed us to write better documentation regarding bandpass construction!

**Comment by ziotom78** [comment]
Hi @nraffuzz and @sgiardie , is this PR ready to be merged? I can update the CHANGELOG and do it, just please tell me if there are showstoppers.

**Comment by nraffuzz** [comment]
No showstoppers from my side, I believe this PR can be merged!
Thanks :)

**Comment by sgiardie** [comment]
Yes, I think it can be merged now!

**Comment by sgiardie** [comment]
(sorry, I just added a tiny correction to the notebook, now should be perfect!)

**Pull request merged**